src/test/java/gov/cabinetoffice/gap/service/ZipServiceTest.java

```diff
@@ -120,15 +120,37 @@ void createZip_onlyTwoFilesCreated() throws Exception {
     }
 
     @Test
-    void shouldReturnLastPartOfSubStringWhenFileNameIsParsed() {
+    void shouldHandleMultiplePeriodsInFilename() {
+        String result = ZipService.parseFileName("330/submission/folder/file.odt.w..pdf", 1, "330","submission");
 
-        String result = ZipService.parseFileName("s3://gap-attachments" +
-                "/13" +
-                "/035f31f6-2877-4f87-8a9b-32649a527668" +
-                "/9d4a3ed6-5a26-4d02-be49-bab13c11a2bf" +
-                "/test File Name.odt", 1);
+        assertEquals("file.odt.w._1.pdf", result);
+    }
+    @Test
+    void shouldHandleFileNameThatAreNotInTheRegex() {
+        String result = ZipService.parseFileName("330/submission/folder/file.word", 1, "330","submission");
+
+        assertEquals("file_1.word", result);
+    }
+
+    @Test
+    void shouldHandleFileNameThatAreNotInTheRegexWithMoreDots() {
+        String result = ZipService.parseFileName("330/submission/folder/file.a.b.c.d.word", 1, "330","submission");
+
+        assertEquals("file.a.b.c.d_1.word", result);
+    }
+
+    @Test
+    void shouldReturnFileNameWithSuffix() {
+        String result = ZipService.parseFileName("330/submission/folder/file.pdf", 1, "330","submission");
+
+        assertEquals("file_1.pdf", result);
+    }
+
+    @Test
+    void shouldReturnFileNameWithSuffixWhenFileNameHasLoadsOfSpecialCharacter() {
+        String result = ZipService.parseFileName("330/submission/folder//test... /File {{{}}} Name.???FLL. odt.&*%*$odt.xls", 1, "330","submission");
 
-        assertEquals("test File Name_1.odt", result);
+        assertEquals("/test... /File {{{}}} Name.???FLL. odt.&*%*$odt_1.xls", result);
```

Comment:
A lot of these symbols we would want to strip out - some regex I've used looks like: `/^[\w\-. ]+$/`

Reply:
extra slashes, question marks in particular but lets be safe and strip other ones too

Reply:
i raised ticket to do that in the backend.. we don't want those special character neither in the s3 bucket.. so there will be a bit of consistency on the file name between json blob, s3, and export

Reply:
https://technologyprogramme.atlassian.net/browse/GAP-2226

Reply:
changed the regex, but i addedd it